types/type-provider.d.ts

```diff
@@ -93,3 +93,11 @@ RouteGeneric
 // review: support both async and sync return types
 // (Promise<Return> | Return | Promise<void> | void)
   : unknown
+
+/**
+ * This branded type is needed to please `typescript-eslint`.
```

Comment:
I protest this phrasing 😄. typescript-eslint is just one tool for typed linting that happens to be the only popular one right now. Soon, others such as Biome, deno lint, and oxc will have equivalent `no-floating-promises` rules. Those rules will likely by default also report on the same situations that [`@typescript-eslint/no-floating-promises`](https://typescript-eslint.io/rules/no-floating-promises) reports.

In other words, this `SafePromiseLike` is needed _for the code to explain itself_, not because any one particular external tool has its own opinions.

```suggestion
 * This branded type is needed to indicate APIs that return Promise-likes which can
 * safely "float" (not have rejections handled by calling code).
```

Reply:
Updated on [0b9de79](https://github.com/fastify/fastify/pull/5506/commits/0b9de798b62f53a6cebe09e4272ca3906efcf905) ✅